src/main/java/com/familytime/controller/BonusController.java

```diff
@@ -0,0 +1,240 @@
+import com.familytime.model.entity.Family;
+import com.familytime.model.service.BonusService;
+import com.familytime.model.service.SecurityService;
+
+import com.familytime.view.form.BonusForm;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+import javax.servlet.http.HttpServletResponse;
+import javax.validation.Valid;
+import java.util.List;
+
+/**
+ * Controller for bonus.
+ *
+ * @version 1.0
+ */
+@RestController
+@RequestMapping (value = "/bonus")
+public class BonusController {
+    /// *** Properties  *** ///
+
+    /**
+     * Service for work with bonus.
+     */
+    @Autowired
+    protected BonusService bonusService;
+
+    /**
+     * Service for getting data from logged user.
+     */
+    @Autowired
+    protected SecurityService securityService;
+
+    /// ***   Methods   *** ///
+
+    /**
+     * Get bonus by id.
+     *
+     * @param id          Id of Bonus.
+     * @param response    Use for work with HTTP.
+     * @return            Found Bonus.
+     */
+    @RequestMapping( value = "/{id}", method = RequestMethod.GET )
+    @ResponseBody
+    public Bonus findById(
+            @PathVariable( "id" )
+            Long id,
+
+            HttpServletResponse response
+    ) {
+        try {
+            //- Search requested bonus -//
+            Bonus bonus = this.bonusService.findById(id);
+            //- Check if bonus was found -//
+            notNull(bonus);
+            return bonus;
+        } catch ( IllegalArgumentException e ) {
+            //- Error. Cannot find this bonus type -//
+            response.setStatus( HttpServletResponse.SC_NOT_FOUND );
+        }
+
+        return null;
+    }
+
+    /**
+     * Get list of all bonuses.
+     * @return List of all bonuses.
+     */
+    @RequestMapping(method = RequestMethod.GET)
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public List<Bonus> findAll() {
+        return this.bonusService.findAll();
+    }
+
+    /**
+     * Get list of bonuses by Family.
+     * @return List of bonuses by Family.
+     */
+    @RequestMapping(method = RequestMethod.GET)
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public List<Bonus> findByFamily() {
+
+        //- Get user's family -/
+        Family usersFamily = securityService.getFamilyOfLoggedUser();
+        return this.bonusService.findByFamily(usersFamily);
+    }
+
+    /**
+     * Get list of bonuses by Bonus type.
+     * @return List of bonuses by Bonus type.
+     */
+    @RequestMapping(method = RequestMethod.GET)
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public List<Bonus> findByBonusType(
+            @RequestBody
+            @Valid
+            BonusForm bonusForm
+    ) {
+        try {
+            if (bonusForm == null) {
```

Comment:
You will never get bonusForm as a `null` here.
See comment above. You shoul use GET params.